common/src/main/java/com/joyent/http/signature/KeyFingerprinter.java

```diff
@@ -0,0 +1,257 @@
+/*
+ * Copyright (c) 2017, Joyent, Inc. All rights reserved.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.joyent.http.signature;
+
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.security.KeyPair;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.security.PublicKey;
+import java.security.interfaces.DSAPublicKey;
+import java.security.interfaces.ECPublicKey;
+import java.security.interfaces.RSAPublicKey;
+import java.util.Base64;
+import java.util.Objects;
+import javax.xml.bind.DatatypeConverter;
+
+/**
+ * Utiility class for calculating and verifying SSH key fingerprints as defined by OpenSSH.
+ */
+public final class KeyFingerprinter {
+
+    /** OpenSSH does not pad.
+     */
+    private static Base64.Encoder b64Encoder = Base64.getEncoder().withoutPadding();
+
+    @SuppressWarnings("checkstyle:javadocmethod")
+    private KeyFingerprinter() {
+    }
+
+    /**
+     * Prior to version 6.7 used an a hex encoded MD5 of the
+     * serialized public key as a fingerprint.  This representation is
+     * still the one used by Triton and Manta as the key id.
+     *
+     * @param keyPair The KeyPair to calculate the fingerprint of
+     * @return The fingerprint (ex: {@code 9f:0b:50:ae:e3:da:f6:eb:b5:71:9a:69:ee:79:9e:c2})
+     */
+    public static String md5Fingerprint(final KeyPair keyPair) {
+        Objects.requireNonNull(keyPair);
+        MessageDigest md = null;
+        try {
+            md = MessageDigest.getInstance("MD5");
+        } catch (NoSuchAlgorithmException e) {
+            // Every implementation of the Java platform is required
+            // to support the following standard MessageDigest
+            // algorithms: MD5, SHA-1, SHA-256
+            throw new AssertionError(e);
+        }
+        byte[] encoded = SshEncoder.encode(keyPair.getPublic());
+        md.update(encoded);
+        byte[] digest = md.digest();
+        return colonify(DatatypeConverter.printHexBinary(digest));
```

Comment:
I only mentioned it because I saw that httpcomponents was a dependency, which then gave us apache codec.

Diving into the common module, I see that we can actually remove the httpcomponents dependency from the common/pom.xml.

I wouldn't change this line. I would remove that unused dependency.